### DISCUSSION TITLE
Apple Pencil doesn’t register as touch onto the canvas for mouse control

### DISCUSSION THREAD

**Discussion by ghovax** [rustdesk/rustdesk#9936] [General]
### Bug Description

When using the Apple Pencil to press the UI buttons it works, but it doesn’t onto the canvas, which doesn’t make sense as the touch actually works there.

### How to Reproduce

Move a finger into the canvas: mouse moves…
Move the Apple Pencil onto the canvas: mouse doesn’t respond!

### Expected Behavior

The Apple Pencil should be able to control the mouse, especially as it provides a much better gliding experience and precise control than fingers do. Also it should be working for clicking, dragging, etc…

### Operating system(s) on local side and remote side

macOS -> iOS

### RustDesk Version(s) on local side and remote side

Latest on macOS -> 1.3.1 iOS

### Screenshots

In the first instant of the video I’m controlling the mouse with my finger, then I’m switching to the Apple Pencil to control it (which doesn’t respond), at last I’m clicking the UI buttons with the pencil, which then works.

https://github.com/user-attachments/assets/2156b3d6-b36c-47c4-a1fc-338c1ffbb63f

**Comment by rustdesk** [comment]
similar topic: 
https://www.reddit.com/r/rustdesk/comments/1gsl7fp/rustesk_and_samsung_pen/
https://github.com/rustdesk/rustdesk/issues/8359

**Comment by MArfanM** [comment]
I would like to fund this, but the Polar.sh page returns a 404.
There is an issue with the funding site it seems

(To note, I am a Samsung user with a Galaxy Tab. I would love to see this turned to reallity!)

**Reply by open-trade** [reply]
You can sponsor via ko-fi, https://ko-fi.com/rustdesk, and then mention here.

**Reply by MArfanM** [reply]
Noted, done! 
@open-trade @ghovax **Funded on Ko-Fi for 30**

**Reply by rp1231** [reply]
What amount does it need to  be funded by to be implemented?
Thanks

**Comment by harumscarum** [comment]
Same here. Apple pensil not working  so forced to use AnyDesk instead

**Comment by fufesou** [comment]
Hi @MArfanM   Could you please try this build?  https://github.com/fufesou/rustdesk/releases/tag/test-stylus

**Reply by eitchtee** [reply]
The Samsung S Pen works nicely in this build.

I've noticed there's no easy way to scroll using one, not a bug, just a flaw in the current control map.

**Reply by MArfanM** [reply]
> Hi @MArfanM   Could you please try this build?  https://github.com/fufesou/rustdesk/releases/tag/test-stylus

Hi, thanks for working on this!

Apologies on not responding sooner, I'll give this a try as soon as I can.

Based on Eichtee's response, it looks to work pretty well. Can't wait to try it!

**Reply by rp1231** [reply]
@fufesou Would it be possible for you to support apple pencil as well?
Thanks

**Reply by fufesou** [reply]
@rp1231 Hi, could you please try the nightly build? Apple pencil and Android pencil may be fixed with the same changes.

**Reply by rp1231** [reply]
@fufesou I  can move the cursor around by hovering the pencil, but I'm unable to tap to click.
Also  please add pressure sensitivity, tilt support and gesture support if possible.
If it's not a lot of work.
Thanks

**Reply by fufesou** [reply]
@rp1231 Sorry, if you're using the iOS app, you may need to wait for an update from TestFlight.

**Reply by rp1231** [reply]
@fufesou The testflight is full.
Would it be possible to get an Ipa file for testing?

**Reply by fufesou** [reply]
@rp1231 Sorry, not for now.

**Reply by MArfanM** [reply]
Hi @fufesou 

First of all, thank you all for working on this feature!

I have the nightly build of Rustdesk running on my laptop and tablet.
I opened an art app to test the stylus behavior and wonder about the behavior.

I am using the Samsung Galaxy S10 Ultra connected to Windows 10 - build 2025-07-23.

The stylus input seems to work exactly like a touch input. In other words, the app treats stylus input no different to touch input (in art apps, touch input is typically ignored for drawing) and there is no pressure sensitivity information transmitted to the Windows device.

I wonder if this is something you noticed too @eitchtee ? I hope I'm using the right build.

Additionally, something not directly related to styluses:

It seems that multi-touch gestures are not transmitted to the device as well. For instance, in art apps, pinch zooming doesn't work (it pinches the rustdesk viewport instead) and 'two-finger-tap' gestures (i.e. tapping two fingers on the screen simultaneously) doesn't seem to work either. The latter (as well as three-finger-tapping among others) is a common gesture for art apps, so they'd be a huge plus to have.

Moreover, this is a connection-related issue, but it seems multiple strokes drawn quickly one after another are often lost. This could get problematic when drawing, so I hope this could be improved upon with a buffer.

Thanks again!

**Reply by eitchtee** [reply]
> I wonder if this is something you noticed too @eitchtee ? I hope I'm using the right build.

Yes, that seems about right. I'm not an artist so I didn't go very deep on testing as you did, just tested basic point and click input which works, but is very limited as I said, no scrolling, etc.

As a matter of fact, Android input is somewhat unreliable, I've got a bluetooth keyboard with a trackpad connected to my tablet, while the keyboard works, the trackpad refuses to register any gestures besides left click and basic movement, two finger click for right mouse is registed as a Left click, two finger scroll is registered as a left click and so on, which borders the unusable and made me change software for the time being, which is a shame as I adore Rustdesk.
